pkg/kube/sourcerepositories.go

```diff
@@ -39,13 +40,40 @@ func GetRepositoryGitURL(s *v1.SourceRepository) (string, error) {
 	return spec.HTTPCloneURL, nil
 }
 
-// FindSourceRepository returns a SourceRepository for the given namespace, owner and name
-// or returns an error if it cannot be found
-func FindSourceRepository(jxClient versioned.Interface, ns string, owner string, name string) (*v1.SourceRepository, error) {
+// FindSourceRepository returns a SourceRepository for the given namespace, owner, repo name, and (optional) provider name.
+// If no SourceRepository is found, return an error if doNotFailIfNotFound is false, otherwise just return a nil repo.
+func FindSourceRepository(jxClient versioned.Interface, ns string, owner string, name string, providerName string, doNotFailIfNotFound bool) (*v1.SourceRepository, error) {
 	resourceName := naming.ToValidName(owner + "-" + name)
+	var errToReturn error
 	repo, err := jxClient.JenkinsV1().SourceRepositories(ns).Get(resourceName, metav1.GetOptions{})
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to find SourceRepository %s in namespace %s", resourceName, ns)
+		if apierrors.IsNotFound(err) {
+			// Wipe the repo, since Get returns an empty SourceRepository, not a nil one, if not found
```

Comment:
why not extract this into a new method `FindSourceRepositoryByLabel`. The function gets shorter and easier to understand and I think a  `FindSourceRepositoryByLabel` makes sense in its own right. Not to mention that you could test `FindSourceRepositoryByLabel` now in isolation.

Reply:
Okiedokie.